Jenkinsfile

```diff
@@ -0,0 +1,225 @@
+                              // Clone repository with terraform
+                              git (branch: tfCodeBranch, url: tfCodeRepo)
+                              dir ("terraform/aws") {
+                                // Run TF Init
+                                if (terraformInit(TF_STATE_BUCKET, tfProject, tfEnvironment, tfComponent, tfRegion) !=0) { error("Terraform init failed")}
+
+                                // Run TF Plan
+                                // Change echo to be error once the AWS has been set up
+                                if (terraform('plan', TF_STATE_BUCKET, tfProject, tfEnvironment, tfComponent, tfRegion, tfVariables) !=0 ) { echo("Terraform Plan failed")}
+
+                                //Run TF Apply
+                                // Change echo to be error once the AWS has been set up
+                                if (terraform('apply', TF_STATE_BUCKET, tfProject, tfEnvironment, tfComponent, tfRegion, tfVariables) !=0 ) { echo("Terraform Apply failed")}
+                              } // dir terraform/aws
+                            } // dir integration-adaptors
+                        } //script
+                    } //steps
+                } //stage
+            } //stages
+        }
+
+
+    }
+    post {
+        always {
+            sh label: 'Stopping containers', script: 'docker-compose down -v'
+            sh label: 'Remove all unused images not just dangling ones', script:'docker system prune --force'
+            sh 'docker image rm -f $(docker images "*/*:*${BUILD_TAG}" -q) $(docker images "*/*/*:*${BUILD_TAG}" -q) || true'
+        }
+    }
+}
+
+int terraformInit(String tfStateBucket, String project, String environment, String component, String region) {
+  println("Terraform Init for Environment: ${environment} Component: ${component} in region: ${region} using bucket: ${tfStateBucket}")
+  String command = "terraform init -backend-config='bucket=${tfStateBucket}' -backend-config='region=${region}' -backend-config='key=${project}-${environment}-${component}.tfstate' -input=false -no-color"
+  dir("components/${component}") {
+    return( sh( label: "Terraform Init", script: command, returnStatus: true))
+  } // dir
+} // int TerraformInit
+
+int terraform(String action, String tfStateBucket, String project, String environment, String component, String region, Map<String, String> variables=[:], List<String> parameters=[]) {
+    println("Running Terraform ${action} in region ${region} with: \n Project: ${project} \n Environment: ${environment} \n Component: ${component}")
+    variablesMap = variables
+    variablesMap.put('region',region)
+    variablesMap.put('project', project)
+    variablesMap.put('environment', environment)
+    variablesMap.put('tf_state_bucket',tfStateBucket)
+    parametersList = parameters
+    parametersList.add("-no-color")
+    //parametersList.add("-compact-warnings")  /TODO update terraform to have this working
```

Comment:
Is this something we plan to do? I noticed it's also commented out in the NHAIS project

Reply:
@bartek-sarul do we have any plans about updating the terraform? Or it's ok to just remove it?

Reply:
As per @michael-kainos's advice, we can remove this

Reply:
Has been fixed in c00dbc4b58360efd8151c61f292e624c1f44fcd4